### PR TITLE
Add hotkey saving functionality

### PR DESCRIPTION
**Pull request by kevinshen1101**

### CODE REVIEW

File: extensions/default/src/ViewerLayout/index.jsx [hunk 1]
```diff
@@ -95,6 +96,13 @@ function ViewerLayout({
   viewports,
   ViewportGridComp,
 }) {
+  hotkeysManager.getHotkeys()
+      .then(res => {
+        console.log("KEVIN RES");
```
Comment (bigshoesdev) [inline extensions/default/src/ViewerLayout/index.jsx:101]: Remove console log

File: extensions/default/src/ViewerLayout/index.jsx [hunk 2]
```diff
@@ -119,6 +127,8 @@ function ViewerLayout({
         onSubmit: ({ hotkeyDefinitions, language }) => {
           i18n.changeLanguage(language.value);
           hotkeysManager.setHotkeys(hotkeyDefinitions);
+          hotkeysManager.saveHotkeys(hotkeyDefinitions);
+          console.log(hotkeyDefinitions);
```
Comment (bigshoesdev) [inline extensions/default/src/ViewerLayout/index.jsx:131]: Remove console log

File: platform/core/src/classes/HotkeysManager.js [hunk 3]
```diff
@@ -63,9 +64,9 @@ export class HotkeysManager {
    */
   setHotkeys(hotkeyDefinitions = []) {
     try {
+      // console.log(hotkeyDefinitions)
```
Comment (bigshoesdev) [inline platform/core/src/classes/HotkeysManager.js:67]: Remove console log

File: platform/core/src/classes/HotkeysManager.js [hunk 4]
```diff
@@ -76,6 +77,49 @@ export class HotkeysManager {
...
+        title: 'Hotkeys Manager',
+        message: 'Error while getting hotkeys',
+        type: 'error',
+      });
+    }
+  }
+
+  async saveHotkeys(hotkeyDefinitions = []) {
+    try {
+      // console.log(hotkeyDefinitions);
```
Comment (bigshoesdev) [inline platform/core/src/classes/HotkeysManager.js:101]: Remove console log

File: platform/core/src/classes/HotkeysManager.js [hunk 5]
```diff
@@ -76,6 +77,49 @@ export class HotkeysManager {
...
+      const hotKeysResponse = await nlApi.put("/api/hotkeys/$self/",
+        {hotkeys: {
+          hotkeyDefinitions,
+        }},
+      );
+      if(hotKeysResponse.status !== 200){
```
Comment (bigshoesdev) [inline platform/core/src/classes/HotkeysManager.js:107]: If we have an error from the backend, does it come into catch closure?

File: extensions/default/src/ViewerLayout/index.jsx [hunk 6]
```diff
@@ -95,6 +96,13 @@ function ViewerLayout({
   viewports,
   ViewportGridComp,
 }) {
+  hotkeysManager.getHotkeys()
+      .then(res => {
+        console.log("KEVIN RES");
+        hotkeysManager.setHotkeys(res.hotkeys.hotkeyDefinitions);
+        console.log(res.hotkeys.hotkeyDefinitions);
```
Comment (bigshoesdev) [inline extensions/default/src/ViewerLayout/index.jsx:103]: Remove console log

File: platform/core/src/classes/HotkeysManager.js (hunk 5 above)
Comment (kevinshen1101) [reply platform/core/src/classes/HotkeysManager.js:107]: Yes, it does. I tested by changing the response code here, and the catch block does execute

**Review by bigshoesdev — APPROVED** [review]

**Pull request merged**